bench/gen_synthetic.ml

```diff
@@ -0,0 +1,104 @@
+open Printf
+
+let dir_rows = ref 1
+
+let dir_cols = ref 1
+
+let mod_rows = ref 1
+
+let mod_cols = ref 1
+
+let comment_size = ref 5000
+
+let count n = Array.to_list (Array.init n (fun k -> k + 1))
+
+let write_modules basedir dir_row dir_col =
+  for row = 1 to !mod_rows do
+    for col = 1 to !mod_cols do
+      let deps =
+        if row = 1 then
+          if dir_row = 1 then []
+          else
+            List.flatten
+              (List.map
+                 (fun k ->
+                   List.map
+                     (fun j ->
+                       sprintf "M_%d_%d_%d_%d.f()" (dir_row - 1) j !mod_rows k)
+                     (count !dir_cols))
+                 (count !mod_cols))
+        else
+          List.map
+            (fun k -> sprintf "M_%d_%d_%d_%d.f()" dir_row dir_col (row - 1) k)
+            (count !mod_cols)
+      in
+
+      let deps = List.rev ("()" :: List.rev deps) in
+
+      let str_deps = String.concat ";\n  " deps in
+      let comment = String.make !comment_size 'X' in
+      let mod_text = sprintf "(* %s *)\nlet f() =\n  %s\n" comment str_deps in
+      let modname =
+        sprintf "%s/m_%d_%d_%d_%d" basedir dir_row dir_col row col
+      in
+      let f = open_out (sprintf "%s.ml" modname) in
+      output_string f mod_text;
+      close_out f;
+      let f = open_out (sprintf "%s.mli" modname) in
+      output_string f "val f : unit -> unit ";
+      close_out f
+    done
+  done;
+  let f = open_out (sprintf "%s/main.ml" basedir) in
+  let modname =
+    sprintf "Test.M_%d_%d_%d_%d" dir_row dir_col !mod_rows !mod_cols
+  in
+  let main = sprintf "let () = %s.f()" modname in
+  output_string f main;
+  close_out f
+
+let dune =
+  {|
+(library
+ (name test)
+ (modules :standard \ main))
+
+(executable
+ (name main)
+ (libraries test)
+ (modules main))
+|}
+
+let write basedir =
+  let () = Unix.mkdir basedir 0o777 in
+  let f = open_out (Filename.concat basedir "dune") in
+  output_string f dune;
+  let () = close_out f in
+
+  for row = 1 to !dir_rows do
+    for col = 1 to !dir_cols do
```

Comment:
Why do we need dir_rows and dir_cols? I would expect a parameter for the number of modules and that's it